### ISSUE TITLE
spec_args schema does not recognize required_if attribute

### ISSUE THREAD

**Issue by nka11** [ansible/ansible-lint#3241]
##### Summary

spec_args schema does not recognize required_if attribute

schema rule fails

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.14.4 using ansible 2.13.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

create a meta/argument_specs.yml file with the required_if option

```console (paste below)
WARNING  Listing 1 violation(s) that are fatal
schema[arg_specs]: Additional properties are not allowed ('required_if' was unexpected)
roles/document_opcon_schedules/meta/argument_specs.yml:1  Returned errors will not include exact line numbers, but they will mention
the schema name being used as a tag, like ``schema[playbook]``,
``schema[tasks]``.

This rule is not skippable and stops further processing of the file.

If incorrect schema was picked, you might want to either:

* move the file to standard location, so its file is detected correctly.
* use ``kinds:`` option in linter config to help it pick correct file type.
```

##### Desired Behavior

Implement latest argument_specs schema

##### Actual Behavior

Fails as described above

**Comment by ssbarnea** [comment]
@cidrblock Let me know if you have an idea about what we can do to ensure that new features are not added without also having the schemas updated. If code would have test files that make use of these key keywords, we could use the linter to ensure that is passing on these.

**Comment by s-hertel** [comment]
`required_*` fields are only supported for module argspec code, not roles. This was a design decision for the initial feature, so `required_if` would still need to be implemented. The documentation lists the valid fields currently.

Using unsupported fields are ignored (for the most part), which I am fixing in https://github.com/ansible/ansible/pull/79657.

**Comment by cidrblock** [comment]
If the sample specification was updated to reflect all the acceptable keys, we could grab that and test against it.  But there would need to be a way to ensure it was always updated.

Testing against it might still be a good idea though since we all would know it was at least passing, and if we miss one in the future, we could PR the sample update. Grabbing the rst and pulling it out shouldn't too bad.

**Comment by ssbarnea** [comment]
@cidrblock We should never have to parse a RST file to check for document fields because newly added features should never be implemented without having fixtures to test that they work. it would be enough for us to lint the fixtures to spot schema issues.

If needed, we can release a new version of the linter every day, so we will never prevent addition of a new features to ansible due to its key being missing from the schema.
